### DISCUSSION TITLE
I have problem with IRequestPreProcessor pipeline

### DISCUSSION THREAD

**Discussion by shirin-monzavi** [LuckyPennySoftware/MediatR#1006] [Q&A]
Hello,
I have two problems.
Firstly,
 I need to write log and my logger is specific for us.

` private readonly ILoggerProvider<GetBalanceQueryHandler> _logger;`

GetBalanceQueryHandler is type of the handler and i want to use IRequestPreProcessor.

when i inject ILoggerProvider<GetBalanceQueryHandler> I cannot get the type of handler.

Note: GetBalanceQueryHandler is a sample and for every scenario is different.

Secondly,
I want to write some log for specific handlers how could i deal with?

I will look forward hearing you.

I really appreciate you in advance.

**Comment by jbogard** [comment]
I don't think you can get specifically the handler type. There's not really an easy way with generics to know the exact handler type resolved by the container. The best you can do is do something like `ILoggerProvider<IRequestHandler<TRequest, TResponse>>`.

Otherwise you'll have to inject the original handler, then use some kind of logger factory to create a logger provider based on the handler getting injected.

**Reply by shirin-monzavi** [reply]
I really appreciate you. but it would be greater if it was possible.
